hcl_to_json.go

```diff
@@ -0,0 +1,230 @@
+// Adapted from https://github.com/tmccombs/hcl2json
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/zclconf/go-cty/cty"
+	ctyconvert "github.com/zclconf/go-cty/cty/convert"
+	ctyjson "github.com/zclconf/go-cty/cty/json"
+)
+
+type jsonObj map[string]interface{}
+
+// Convert an hcl File to a json serializable object
+// This assumes that the body is a hclsyntax.Body
+func convertFile(file *hcl.File) (jsonObj, error) {
+	c := converter{bytes: file.Bytes}
+	body := file.Body.(*hclsyntax.Body)
+	return c.convertBody(body)
+}
+
+type converter struct {
+	bytes []byte
+}
+
+func (c *converter) rangeSource(r hcl.Range) string {
+	data := string(c.bytes[r.Start.Byte:r.End.Byte])
+	data = strings.ReplaceAll(data, "\n", "")
+	data = strings.ReplaceAll(data, " ", "")
+	return data
+}
+
+func (c *converter) convertBody(body *hclsyntax.Body) (jsonObj, error) {
+	var err error
+	out := make(jsonObj)
+	for key, value := range body.Attributes {
+		out[key], err = c.convertExpression(value.Expr)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	for _, block := range body.Blocks {
+		err = c.convertBlock(block, out)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return out, nil
+}
+
+func (c *converter) convertBlock(block *hclsyntax.Block, out jsonObj) error {
+	var key string = block.Type
+
+	value, err := c.convertBody(block.Body)
+	if err != nil {
+		return err
+	}
+
+	for _, label := range block.Labels {
+		if inner, exists := out[key]; exists {
+			var ok bool
+			out, ok = inner.(jsonObj)
+			if !ok {
+				// TODO: better diagnostics
```

Comment:
@lbn is this our own issue, or borrowed from original tmccombs/hcl2json?

Reply:
Copy and pasted.